### PR TITLE
Sergey cheremshinsky

### PR DESCRIPTION
**Pull request by sergTch**
<h1>Description</h1>

Start - Preprocessing data "spliting columns. Example:
 Embark that has values S, Q, C splited into 
 Embark_S, Embark_Q, Embark_C with values 0 or 1

Train - made function that train 5 models from sklearn and allow them to vote 
 on validation test showes ~82.5% accuracy
 also made neural network using tensorflow with ~86% accuracy

However both tries showes only 76-80% accuracy on kaggle

<h2>How has this been tested?</h2>

- [ ] Executed code few times from start to end without crashing
- [ ] Only ran once
- [ ] Code have dedicated unit tests

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+data = preprocess(data)
+x, y = split_data(data)
+
+a = []
+for i in range(1):
+    x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2)
+    x_train = x
+    y_train = y
+    # exit()
```
Comment (meanalexrin) [inline main.py]: Usually it's better to remove unused code from the project. It helps readability

File: main.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    submission = pd.DataFrame({
+        "PassengerId": test_ids,
+        "Survived": y
+    })
+    submission.to_csv('./titanic.csv', index=False)
+    print('Exported!')
+
+
+generateAns()
```
Comment (meanalexrin) [inline main.py]: You may consider using `if __name__ == "__main__"` in the python code to indicate entry points to the program 

https://stackoverflow.com/a/419185

**Review by meanalexrin — CHANGES_REQUESTED** [review]
LGTM - Looks Good To Me
Except:
Please move you work to the `/[username]_code` location, as described in readme
https://github.com/rnd4u-org/2021-knu-cairl#development

I'll approve after that
Also that's a good approach that you are using separate .py files rather than a single jupyter notebook

**Comment by sergTch** [comment]
I fixed placements of files. Added solution for mask classification.
Didn't add to titanic solution if _name_ == "_main_": but used it in masks
In new project also deleted all code comments however regretted few times. Need to get used to it:)

**Comment by sergTch** [comment]
On masks dataset used inception v3 with 1024 512 and 128 dense layers. Trained in 3 steps
freezing 0 : 210 layers, then 0 : 105 and 210 : 316 and last freezed 105 : 316
that helped to train all parts of cnn and achieve 100% on training data and 99.4% on test
At start splited data on test and train where test is 20% of dataset and train is other 80%

File: sergey_cheremshinsky/task3(mask)/load_data.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    size = 224
+
+    try:
+        x_train = np.load('x_train.npy', allow_pickle=True)
+        x_test = np.load('x_test.npy', allow_pickle=True)
+        y_train = np.load('y_train.npy', allow_pickle=True)
+        y_test = np.load('y_test.npy', allow_pickle=True)
+
+        return (x_train, x_test, y_train, y_test)
+    except Exception:
```
Comment (meanalexrin) [inline sergey_cheremshinsky/task3(mask)/load_data.py]: Try to use less general errors or if statements

File: sergey_cheremshinsky/task3(mask)/main.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        for i in range(size):
+            model.layers[i].trainable = size - size / steps * step > i and size - size / steps * (step + 1) >= i
+            model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
+            history = model.fit(
+                x=x_train, 
+                y=y_train,
+                epochs=10,
+                validation_data=(x_test, y_test),
+                validation_steps=1511 // batch,
+                steps_per_epoch=6042 // batch,
```
Comment (meanalexrin) [inline sergey_cheremshinsky/task3(mask)/main.py]: Please add constants and description for magical numbers like 1511 and 6042

File: sergey_cheremshinsky/task3(mask)/load_data.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            ind += 1
+
+    shuffle(indexes)
+    n = len(images) * 4 // 5
+
+    x_train = [images[i] for i in indexes[:n]]
+    x_test = [images[i] for i in indexes[n:]]
+
+    y_train = [labels[i] for i in indexes[:n]]
+    y_test = [labels[i] for i in indexes[n:]]
```
Comment (meanalexrin) [inline sergey_cheremshinsky/task3(mask)/load_data.py]: Please use existing train test split function to assure correct execution

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

**Review by meanalexrin — CHANGES_REQUESTED** [review]
Please check other comments

Regarding task 3:
You can try rerun model few times without saving `.npy` indexes. So you'll test model on few different train/test splits

**Comment by sergTch** [comment]
I used saving of model and saving of splited data mostly because that's hard for my laptop to finish training in some reasonable amount of time. So I splited training for three nights saving model after each and using same split of data

**Comment by sergTch** [comment]
I fixed mistakes that you mentioned about task 3 and also pushed solution for task 4
